### DISCUSSION TITLE
Java Compiler source

### DISCUSSION THREAD

**Discussion by happytomatoe** [DavidSouther/computron5k#482] [General]
@DavidSouther @Shimon-Schocken I am adding new validation rules to  the web ide  compiler and would like to port  some of the validations from the old Java compiler. Do you happen to know where are the sources for old Java compiler? I don't see them inside https://github.com/nand2tetris/nand2tetris_simulator

**Comment by DavidSouther** [comment]
There are no officially provided Jack compiler source code, as that would circumvent the purpose of the final project in the course being to write said compiler. @Shimon-Schocken  may have the original choice available.
